src/bp/core/services/ghost/service.ts

```diff
@@ -266,6 +257,33 @@ export class ScopedGhostService {
     return allFiles
   }
 
+  public async importFromDirectory(directory: string) {
+    const filenames = await this.diskDriver.absoluteDirectoryListing(directory)
+
+    const files = filenames.map(file => {
+      return {
+        name: file,
+        content: fse.readFileSync(path.join(directory, file))
+      } as FileContent
+    })
+
+    await this.upsertFiles('/', files)
+  }
+
+  public async exportToArchiveBuffer(): Promise<Buffer> {
+    const tmpDir = tmp.dirSync({ unsafeCleanup: true })
+
+    try {
+      const outFiles = await this.exportToDirectory(tmpDir.name)
+      const filename = path.join(tmpDir.name, 'archive.tgz')
+
+      const archive = await createArchive(filename, tmpDir.name, outFiles)
+      return await fse.readFile(archive)
+    } finally {
```

Comment:
what if an error occurs here ? don't you want to log smth? or is the logging handled in within `exportToDirectory` and `createArchive` funcs?

Reply:
Other methods all throws an exception, so it would be the same here. But yeah it could be added